src/provider/provider_level_zero.c

```diff
@@ -347,12 +354,29 @@ static umf_result_t ze_memory_provider_open_ipc_handle(void *provider,
     ASSERT(providerIpcData != NULL);
     ASSERT(ptr != NULL);
     ze_result_t ze_result;
-    ze_ipc_mem_handle_t *ze_ipc_handle = (ze_ipc_mem_handle_t *)providerIpcData;
+    ze_ipc_data_t *ze_ipc_data = (ze_ipc_data_t *)providerIpcData;
     struct ze_memory_provider_t *ze_provider =
         (struct ze_memory_provider_t *)provider;
+    int fd_local = -1;
+    ze_ipc_mem_handle_t ze_ipc_handle = ze_ipc_data->ze_handle;
+
+    if (ze_ipc_data->pid != utils_getpid()) {
+        int fd_remote = -1;
+        memcpy(&fd_remote, &ze_ipc_handle, sizeof(fd_remote));
```

Comment:
Using memcpy to copy single int is an overkill

Reply:
Compiler optimizes it. When the size is low enough there is now an actual call to memcpy, it is replaced with move instructions.

Reply:
It's more code readability issue then performance. Assignment is shorter, and easier to read.

Reply:
regarding readability, I am not sure:

```cpp
fd_remote = *(int*)&ze_ipc_handle;
```
vs
```cpp
memcpy(&fd_remote, &ze_ipc_handle, sizeof(fd_remote));
```

and

```cpp
*((int*)&ze_ipc_handle) = fd_local;
```
vs
```cpp
memcpy(&ze_ipc_handle, &fd_local, sizeof(fd_local));
```

Are you still suggesting to get rid of memcpy? If so, I will do that. Please confirm.

Reply:
all this is needed just because L0 IPC implementation is broken.
According to the L0 spec all these steps not needed at all. When L0 driver fix their implementation we will get rid of these changes.